rfcs/008-map-indexes.md

```diff
@@ -0,0 +1,159 @@
+
+Initially the building of an index will be a single worker running through the changes feed and creating the index. Ideally it would be nice to parallelise that work so that multiple workers could build the index at the same time. This will reduce build times. This can be done by fetching the boundary keys for the changes feed, splitting those key ranges amongst different workers to build different parts of the index. This will require that for each document update processed, the worker must check the revision key space to determine if the document is the latest revision of the document. If it is not it should discard it. The other requirement is that CouchDB could only start serving the index once the update sequence is up to date AND all the workers have completed building their section of the index.
+
+### View clean up
+
+When a design document is changed, new indexes will be built and grouped under a new  `View Signature` . The old map indexes will still be in FDB.  CouchDB will need to run a clean up process that scans all the `View Signature`s and links them to a design document. If a `View Signature` is not linked to any design doc, that `View Signature` namespace can be removed. This will be done via the background jobs scheduler.
+
+### Stale = “ok” and stable = true
+
+ With the consistency guarantee’s CouchDB will get from FDB,  stable = true will no longer be an option that CouchDB would support. Similar `stale = “ok”` would now be translated to `update = false`
+
+### Size limits
+
+* Emitted keys will not be able to exceed 10 KB
+* Values cannot exceed 100 KB
+* There could be rare cases where the number of key-value pairs emitted for a map function could lead to a transaction either exceeding 10 MB in size which isn’t allowed or exceeding 5 MB which impacts the performance of the cluster. Ideally CouchDB will need to detect these situations and split the transaction into smaller transactions
+
+## Advantages
+
+* Map indexes will work on FoundationDB with same behaviour as current CouchDB 2.x
+* Options like stale = “ok” and ‘stable = true’ will no longer be needed
+
+## Disadvantages
+
+* Size limits on key and values
+
+## Key Changes
+
+* Indexes are stored in FoundationDB
+* Indexes will be built via the background job queue
+* ICU sort strings will be generated ahead of time for each key that is a string
+
+## Applications and Modules affected
+
+* couch_mrview will be removed and replaced with a new indexing OTP application
+
+## HTTP API additions
+
+The API will remain the same.
+
+## HTTP API deprecations
+
+No deprecations.
```

Comment:
`stale` and `stable` were mentioned a couple times. Something somehow would be "deprecated" even if its "You'll not get possibly weird data if you elect to do `stale=ok` or so on.

Also reduce functions are something we'll want to call out.

Reply:
I've added in reduce functions. But we will have an RFC for reduce functions later.